src/main/java/org/apache/commons/lang3/ClassUtils.java

```diff
@@ -424,7 +424,14 @@ public static String getAbbreviatedName(final Class<?> cls, final int len) {
      * significant loss of meaning.</p>
      * <p>The abbreviated class name will always include the complete package hierarchy.
      * If enough space is available, rightmost sub-packages will be displayed in full
-     * length.</p>
+     * length. The abbreviated package names will be shortened to a single character.</p>
+     * <p>Only package names are shortened, the class simple name remains untouched. (See examples.)</p>
+     * <p>The result will be longer than the desired length only if all the package names
+     * shortened to a single character plus the class simple name with the separating dots
+     * together are longer than the desired length. In other words, when the class name
+     * cannot be shortened to the desired length without significant loss of meaning.</p>
+     * <p>If the class name can be shortened without significant loss of meaning then
```

Comment:
I do not think we should talk about "loss of meaning" as to me, abbreviating implies "loss of meaning" by design ;-) Just describe what it does. IMO a list would be better to describe the different use cases.